packages/create-sitecore-jss/src/templates/node-xmcloud-proxy/src/personalize.ts

```diff
@@ -0,0 +1,30 @@
+import { PersonalizeHelper } from '@sitecore-jss/sitecore-jss-proxy';
+import { personalizeConfig } from './config';
+import { responseInterceptor } from 'http-proxy-middleware';
+import { Plugin } from 'http-proxy-middleware/dist/types';
+import { IncomingMessageWithBody } from './types';
+export const personalizeHelper = new PersonalizeHelper(personalizeConfig);
+
+// personalize plugin to modify intercepted Layout Service request data
+export const personalizePlugin: Plugin = (proxyServer) => {
+  proxyServer.on(
+    'proxyRes',
+    responseInterceptor(async (responseBuffer, _, req, res) => {
+      let responseText = responseBuffer.toString('utf8');
+      const payload = JSON.stringify((req as IncomingMessageWithBody).body);
+      if (payload.toLowerCase().includes('jsslayoutquery')) {
```

Comment:
We need to introduce a variable GRAPHQL_LAYOUT_QUERY_NAME in "sitecore-jss". It should be referenced value instead of hardcoded